src/pb_stub.cc

```diff
@@ -881,6 +945,10 @@ void
 Stub::Finalize()
 {
   finalizing_ = true;
+  // Stop async event loop if created.
+  if (!py::isinstance<py::none>(async_event_loop_)) {
+    async_event_loop_.attr("stop")();
+  }
```

Comment:
Do we need to wait for the thread to join as well?

Reply:
I don't think it is needed, because the thread is started as daemon and it only does one thing
```python
async_event_loop_.run_forever()
```
which will return after calling
```python
async_event_loop_.stop()
```
and then the thread automatically exits.